src/web/app/pages-instructor/instructor-track-view-page/instructor-track-view-page.component.ts

```diff
@@ -205,50 +210,38 @@ export class InstructorTrackViewPageComponent implements OnInit {
           this.studentService
               .getStudentsFromCourse({ courseId: this.formModel.courseId })
               .subscribe((students: Students) => {
-                students.students.map((student: Student) => this.students.push(student));
+                this.students.push(...students.students);
+
+                const targetFeedbackSessionLog: FeedbackSessionLog | undefined = logs.feedbackSessionLogs
+                  .find((fsLog: FeedbackSessionLog) =>
+                    fsLog.feedbackSessionData.feedbackSessionName === this.formModel.feedbackSessionName);
 
-                logs.feedbackSessionLogs
-                  .filter((fsLog: FeedbackSessionLog) =>
-                    fsLog.feedbackSessionData.feedbackSessionName === this.formModel.feedbackSessionName)[0]
-                    .feedbackSessionLogEntries
+                if (!targetFeedbackSessionLog) {
+                  return;
+                }
+
+                targetFeedbackSessionLog.feedbackSessionLogEntries
                   .filter((entry: FeedbackSessionLogEntry) =>
                     LogType[entry.feedbackSessionLogType.toString() as keyof typeof LogType]
-                      === LogType.FEEDBACK_SESSION_VIEW)
-                  .map((entry: FeedbackSessionLogEntry) => {
-                    if (!(entry.studentData.email in this.studentToLog)
-                      || this.studentToLog[entry.studentData.email].timestamp < entry.timestamp) {
-                      this.studentToLog[entry.studentData.email] = entry;
-                    }
-                  });
-                this.searchResult = this.toFeedbackSessionLogModel(logs.feedbackSessionLogs[0]);
+                      === LogType.FEEDBACK_SESSION_VIEW
+                    && (!(entry.studentData.email in this.studentToLog)
+                        || this.studentToLog[entry.studentData.email].timestamp < entry.timestamp))
```

Comment:
I have a feeling that instead of a few of these logical operators separating the conditions, you could perhaps make them into yet another filter. This feels a bit more functional to me.

Reply:
ok! I agree that using a separate filter will make the code more readable